docs/genqlient.yaml

```diff
@@ -243,6 +243,22 @@ package_bindings:
 
 # Configuration for genqlient's smart-casing.
 #
+# If set to true, genqlient will automatically convert snake_case names to camelCase 
+# before applying the standard capitalization for Go exports. This applies to both 
+# field names and type names. The GraphQL query will still use the original names 
+# for compatibility with the server.
+#
+# This is useful when working with GraphQL servers that use snake_case naming
+# conventions, while maintaining idiomatic Go struct field and type names.
+#
+# For example, with this option enabled:
+# - A GraphQL field named "user_id" will be named "UserId" in the generated Go struct
+# - A GraphQL type named "user_profile" will be named "UserProfile" in Go
+# - All original names are preserved in the GraphQL request
+#
+# Defaults to false.
+auto_camel_case: false
```

Comment:
let's put this under `casing` below? I think the thing that makes sense is to add a `casing.default` with the same values that controls what we do "everywhere", and then support `auto_camel_case` as a value for each. (I'm guessing that's not too much work?) So it would look like
```
casing:
  default: auto_camel_case
```
for what you have, but then someone could also do
```
casing:
  default: auto_camel_case
  all_enums: raw
```
or whatever and have them work together correctly